titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/client/DefaultDirectKubeApiServerIntegrator.java

```diff
@@ -149,6 +154,28 @@ public boolean isReadyForScheduling() {
         return kubeApiFacade.isReadyForScheduling();
     }
 
+    public List<ContainerState> getPodStatus(String taskId) {
+        if (getPods().containsKey(taskId)) {
+            List<V1ContainerStatus> v1ContainerStatus = getPods().get(taskId).getStatus().getContainerStatuses();
+            ArrayList containerstates = new ArrayList();
+            if (v1ContainerStatus.isEmpty()) {
+                // we have pod status but no container status just yet
+                return null;
```

Comment:
Collections.emptylist() is probably safer, also for line 176

Reply:
+1. You always want to return safe default values where you can as `null` is error prone.